src/foam/core/AbstractArrayPropertyInfo.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.core;
+
+import foam.nanos.logger.NanoLogger;
+import java.lang.UnsupportedOperationException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.xml.stream.XMLStreamConstants;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+public abstract class AbstractArrayPropertyInfo
+  extends AbstractPropertyInfo {
+
+  public int compareValues(Object[] b1, Object[] b2) {
+    return (b1 == b2) ? 1 : 0;
+  }
+
+  @Override
+  public void setFromString(Object obj, String value) {
+    // TODO
+  }
+
+  public abstract String of();
+
+  // NESTED ARRAY
+  @Override
+  public Object fromXML(X x, XMLStreamReader reader) {
+    NanoLogger logger = (NanoLogger) x.get("logger");
+    List objList = new ArrayList();
+    String startTag = reader.getLocalName();
+    try {
+      int eventType;
+      while ( reader.hasNext() ) {
+        eventType = reader.next();
+        switch ( eventType ) {
+          case XMLStreamConstants.START_ELEMENT:
+            // Nested object in array
+            if ( reader.getLocalName().equals("object") ) {
+              FObject clsInstance = XMLSupport.createObj(x, reader);
```

Comment:
I find the name 'clsInstance' confusing, since the name would make me suspect it were an instance of a class. Maybe a better/simpler name would just be 'o'. When would this value be null?